### PR TITLE
Issue/8184 remove agent 1

### LINKED ISSUE #8184: Strip out old agent
- [ ] Remove all agent side code only used by the old agent 
- [ ] Make the scheduler start instead of the agent / remove the ability for the agent to start

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Removed the agent.

(Which is very weird, after maintaining it for so long).

1. Most of this is reworking tests, 
2. Test have been re-organised:
  - ~~agent_server~~
  - deploy # all deploy related things
     - e2e # all not mocked deploy things
       -  test_autostarted.py # all test that auto start the agent
       - `test_*.py` per topic files
3. some test have skips, to be investigated later https://github.com/inmanta/inmanta-core/issues/8235

closes #8184 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: changelogs/unreleased/8184_remove_agent.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+description: Removed the agent
+issue-nr: 8184
+change-type: major
+destination-branches: [master]
```
Comment (wouterdb) [inline changelogs/unreleased/8184_remove_agent.yml:4]: Final changelog entry to be added when the epic is done

File: src/inmanta/server/agentmanager.py [hunk 2]
```diff
@@ -1182,7 +1161,7 @@ async def _ensure_agents(
     # Start/Restart scheduler
     async def _ensure_scheduler(
         self,
-        env: data.Environment,
+        env: uuid.UUID,
```
Comment (wouterdb) [inline src/inmanta/server/agentmanager.py:1164]: Updated typing for convenience.

File: src/inmanta/server/agentmanager.py [hunk 3]
```diff
@@ -1257,11 +1238,9 @@ async def __do_start_agent(
         """
         Start an autostarted agent process for the given environment. Should only be called if none is running yet.
         """
-        assert not no_auto_start_scheduler
-
-        use_resource_scheduler: bool = opt.server_use_resource_scheduler.get()
+        assert not assert_no_start_scheduler
```
Comment (wouterdb) [inline src/inmanta/server/agentmanager.py:1241]: purely for tetsing

File: tests/conftest.py [hunk 4]
```diff
@@ -19,15 +19,27 @@
 import logging.config
```
Comment (wouterdb) [inline tests/conftest.py:19]: All fixtures the agent_server subfolder are now here

**Review by bartv — APPROVED** [review]
Looks good. What is the plan with the skipped tests?

**Comment by wouterdb** [comment]
> Looks good. What is the plan with the skipped tests?

https://github.com/inmanta/inmanta-core/issues/8235

File: tests/conftest.py [hunk 5]
```diff
@@ -876,6 +809,87 @@ async def server_multi(
             logger.exception("Timeout during stop of the server in teardown")
 
 
+@pytest.fixture(scope="function")
+async def auto_start_agent():
+    """Marker fixture, indicates we done expected scheduler autostart, any attempt to start the scheduler results in failure"""
```
Comment (arnaudsjs) [inline tests/conftest.py:814]:
```suggestion
    """Marker fixture, indicates we do expect scheduler autostart, any attempt to start the scheduler results in failure"""
```

File: tests/deploy/e2e/test_code_loading.py [hunk 6]
```diff
@@ -244,45 +261,44 @@ async def get_version() -> int:
...
-        environment=environment, agent_map={"agent1": "localhost"}, hostname="host", agent_names=["agent1"], code_loader=True
-    )
-    assert not os.path.exists(os.path.join(agent.executor_manager._loader.mod_dir, "tests"))
+    # # ensure we clean up on restart
+    # assert os.path.exists(os.path.join(executor_manager._loader.mod_dir, "tests/plugins/__init__.py"))
+    # await agent.stop()
+    # await agent_factory(
+    #     environment=environment, agent_map={"agent1": "localhost"}, hostname="host", agent_names=["agent1"], code_loader=True
+    # )
+    # assert not os.path.exists(os.path.join(executor_manager._loader.mod_dir, "tests"))
```
Comment (arnaudsjs) [inline tests/deploy/e2e/test_code_loading.py:289]: What about this?

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: tests/deploy/e2e/test_code_loading.py (hunk 6 above)
Comment (wouterdb) [reply tests/deploy/e2e/test_code_loading.py]: removed thx

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**